### PR TITLE
feat: Add course card context menu on course page

### LINKED ISSUE #80: When I right click on a course card, open a drop down showing several options
### Feature Description

When I right click on a card, ONLY on the Dashboard, show the following options:

1. Clone
2. Share
3. Delete
4. Invite

image

For the UI behavior, use this component: https://carbon-components-svelte.onrender.com/components/ContextMenu

https://github.com/rotimi-best/classroomio/assets/31730715/32d5b627-dd44-4938-ae43-3c0a0a508a7a

### Use Case

_No response_

### Proposed Solution

_No response_

### Alternatives (optional)

_No response_

### Additional Context

_No response_

### Please check the boxes that apply to this feature request.

- [X] I have searched the existing feature requests to make sure this is not a duplicate.
- [X] I have provided a detailed description of the requested feature.
- [X] I have explained the use case or scenario for this feature.
- [X] I have included any relevant technical details or design suggestions.
- [ ] I understand that this is a suggestion and that there is no guarantee of implementation.

### PR DESCRIPTION
**Pull request by tioluwani94**
## What does this PR do?
This PR adds a context menu to the course cards on the "Courses" page when the card is right-clicked. The options added to the menu are:
- Clone
- Share
- Invite
- Delete

Fixes #80 

Screenshot 2023-12-19 at 10 39 17 AM

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?
- On the dashboard, navigate to the "Course" page.
- On the "Course" page, right-click on any course, and you should see a context menu with options to clone the course, share the course, delete the course, and invite people to the course.
- Clicking on any of the options would bring up a browser alert.
- Note the functionality for the options of the menu is not yet implemented.

### Required
- [ ] Filled out the "How to test" section in this PR
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from `main` onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated
- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the ClassroomIO Docs if changes were necessary

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,24 +1,24 @@
 {
```
Comment (rotimi-best) [inline package.json]: I think there is a diff in your prettier config here, not sure why this file changed.

File: apps/dashboard/src/lib/components/Courses/index.svelte [hunk 2]
```diff
@@ -32,6 +32,7 @@
     {#each courses as courseData}
       {#key courseData.id}
         <Card
+          showContextMenu
```
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Courses/index.svelte:35]: Thank you for adding this. You really figured it the right way to implement this so students don't get this option.

**Review by rotimi-best — APPROVED** [review]
Thank you so much @tioluwani94 for this 🙏 😊. Bro you rock.

Apart from the package.json formatting, this looks good to me.

File: package.json (hunk 1 above)
Comment (tioluwani94) [reply package.json]: Yah, I've resolved the issue, thanks @rotimi-best

File: apps/dashboard/src/lib/components/Courses/index.svelte (hunk 2 above)
Comment (tioluwani94) [reply apps/dashboard/src/lib/components/Courses/index.svelte:35]: Thanks @rotimi-best

**Review by rotimi-best — APPROVED** [review]
LGTM

**Pull request merged**